content/1_General/Adding_Solution.mdx

```diff
@@ -88,24 +88,33 @@ See [Working With MDX](/general/working-mdx) for additional information.
     ```
 
     Keep file names and solution IDs consistent; namely, USACO IDs should
-    correspond to the URL on
+    correspond to the last digits at the very end of the URL on
     [usaco.org](http://www.usaco.org/index.php?page=viewproblem2&cpid=690) and
-    file names should be the same as the solution's ID.
+    file names should be the same as the solution's ID. See
+    [Working With MDX](/general/working-mdx#problem-lists) for more examples of
+    IDs.
 
 3.  Add your implementation, following these conventions and the conventions
     [below](#code-conventions):
 
-    - Use _tabs_ to indent.
-    - Lines should not exceed 80 characters.
+    - Use _tabs_ to indent
+    <Info title="Pro Tip">
+	If your code is written with spaces, use the [Guide Editor](https://usaco.guide/editor/), copy-paste your code in, right click, click "Command Palette" from the menu that pops up and then search for "Convert Indentation to Tabs". Once you see it, click on it and then copy that new code with the correct indentation. into your Github pull request.
+    </Info>
+    
+    - Lines should not exceed 80 characters. If you need to, you can write comments across multiple lines to follow this rule.
+    - Your code needs to have comments if it doesn't contribute anything to an already existing solution on the guide or official USACO solution. Comments shouldn't be used for every single obvious line and should only describe complicated or big chunks of code. Comments should be short and concise, being not longer than 2-3 sentences but still encompassing all the valuable information an inexperienced coder might not be able to automatically assume. 
```

Comment:
```suggestion
    - Add comments throughout your code to explain your logic. Feel free to copy-paste the official USACO solution and improve it by adding more descriptive variable names and helpful comments.
```

oops idk what to say about comments; I think it's difficult to create hard rules regarding commenting. I'm open to suggestions

Reply:
> having a list of every trait a good comment encompasses would be helpful.

idk if this is possible...

> Your code needs to have comments

only when it makes sense to have comments, right?

> Comments shouldn't be used for every single obvious line ... Comments should be short and concise

this (should) be obvious...

> being not longer than 2-3 sentences

this is not necessarily always true?

Reply:
you are right hold a vote or something to tell me what to write for the comments guidelines. 
if a) i should include most of what I said reformatted in bullet points so its easier to read or b) commit your changes or c) asking anyone what their idea should be

Reply:
ye ur code doesnt always need comments

comments can be longer than 3 sentences tho rare

comments doesnt have a set rule to follow it just explains code

Reply:
personally i think Nathans changes should be committed

Reply:
one thing i suggest we should add is that comments shouldn't explain what your code does, but instead explain why it does what it does.

Reply:
commited